### PR TITLE
Add support for device registry

### PR DESCRIPTION
**Pull request by lkempf**
Currently this integration only provides entities for the thermostats. I implemented the `device_info` function which causes the thermostats to also show up as devices. More details on the device registry can be found at https://developers.home-assistant.io/docs/device_registry_index

### CODE REVIEW

**Comment by dbuezas** [comment]
Oh this is cool. I intend to try other features like schedules and some diagnostic entities (like connection status, rssi, etc) so this is excellent preparation for that too

File: custom_components/dbuezas_eq3btsmart/climate.py [hunk 1]
```diff
@@ -176,11 +182,6 @@ def available(self) -> bool:
         return True
         return self._thermostat.mode >= 0
 
-    @property
```
Comment (dbuezas) [inline custom_components/dbuezas_eq3btsmart/climate.py:179]: Why did you remove this? Everything is now called Unnamed device
image

Comment (lkempf) [reply custom_components/dbuezas_eq3btsmart/climate.py:179]: I followed the documentation (https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations) but I see the problem. I must have missed that because it works on the dashboard for some reason.

Comment (dbuezas) [reply custom_components/dbuezas_eq3btsmart/climate.py:179]: Ok, I see that adding it makes the name appear twice in different places.
Maybe the name of the entity should be just "Thermostat", then HA would prefix it with the device name and one gets "Living Room Thermostat"

Comment (lkempf) [reply custom_components/dbuezas_eq3btsmart/climate.py:179]: Ah no I see my bug. I set the device name to self.name which is of course none now. It needs to be self._thermostat.name.

Comment (dbuezas) [reply custom_components/dbuezas_eq3btsmart/climate.py:179]: That makes sense.
I already merged it so I'll do the change you mention
